File: src/services/ingestion/extractors/text-extractor.ts
import type { Document } from "../../../types/models.js";
import type { Extractor } from "./base.js";

/** Extractor for plain text documents. */
export class TextExtractor implements Extractor {
  /**
   * Returns raw content as-is.
   * @param doc Source document.
   */
  public async extract(doc: Document): Promise<string> {
    return doc.rawContent;
  }
}

File: src/services/ingestion/extractors/text_extractor.py
from src.types.models import Document

from .base import Extractor


class TextExtractor(Extractor):
    """Extractor for plain text documents."""

    async def extract(self, doc: Document) -> str:
        """Returns raw content as-is.
        
        Args:
            doc: Source document.
            
        Returns:
            Raw content unchanged.
        """
        return doc.raw_content
